### ISSUE TITLE
NPM Audit: Vite dev server option `server.fs.deny` can be bypassed when hosted on case-insensitive filesystem

### ISSUE THREAD

**Issue by tsega** [nuxt/nuxt#25458]
### Environment

- Operating System: Darwin
- Node Version:     v20.10.0
- Nuxt Version:     3.9.3
- CLI Version:      3.10.0
- Nitro Version:    2.8.1
- Package Manager:  npm@10.2.3
- Builder:          -
- User Config:      ssr, devtools, modules, googleFonts
- Runtime Modules:  @nuxtjs/tailwindcss@6.11.1, @nuxtjs/google-fonts@3.1.3, nuxt-lucide-icons@1.0.3, nuxt3-leaflet@1.0.12, @vee-validate/nuxt@4.12.4, nuxt-primevue@0.2.2
- Build Modules:    -

### Reproduction

npm audit report

### Describe the bug

On Nuxt v3.9.3, the suggest solution to `npm audit fix --force` downgrades to v3.8.2 which itself is still affected by this dependency issue.

```
vite  5.0.0 - 5.0.11
Severity: high
Vite dev server option `server.fs.deny` can be bypassed when hosted on case-insensitive filesystem - https://github.com/advisories/GHSA-c24v-8rfc-w8vw
fix available via `npm audit fix --force`
Will install nuxt@3.8.2, which is a breaking change
node_modules/@nuxt/vite-builder/node_modules/vite
```

**Comment by Peggy535** [comment]
I have the same issue and another:

Environment:

- OS = maxOS Sonoma (M2)
- Node v21.6.1
- Nuxt 3.9.3
- Package manager npm 10.2.4

These are my current dependencies:

`"devDependencies": {
		"@nuxt/devtools": "^1.0.8",
		"@nuxtjs/color-mode": "^3.3.2",
		"@pinia-plugin-persistedstate/nuxt": "^1.2.0",
		"@tailwindcss/typography": "^0.5.10",
		"@unlok-co/nuxt-stripe": "^2.0.0",
		"@vueuse/core": "^10.7.0",
		"@vueuse/nuxt": "^10.7.0",
		"nuxt": "3.9.3",
		"nuxt-resend": "^0.0.2",
		"vue": "latest",
		"vue-router": "^4.2.5"
	},
	"dependencies": {
		"@iconify-json/arcticons": "^1.1.87",
		"@iconify-json/ph": "^1.1.10",
		"@nuxt/image": "npm:@nuxt/image-edge@latest",
		"@nuxt/ui": "npm:@nuxt/ui-edge@latest",
		"@nuxtjs/sanity": "^1.8.3",
		"@pinia/nuxt": "^0.5.1",
		"@sanity/client": "^6.9.3",
		"@vue-email/nuxt": "^0.8.11",
		"decimal.js": "^10.4.3",
		"gsap": "^3.12.5",
		"pinia": "^2.1.7",
		"yup": "^1.3.3"
	},`

Running npm audit results with:

`lodash.pick  >=4.0.0
Severity: high
Prototype Pollution in lodash - https://github.com/advisories/GHSA-p6mc-m468-83gw
fix available via `npm audit fix --force`
Will install nuxt@3.8.2, which is a breaking change
node_modules/lodash.pick
    vite-plugin-checker  >=0.2.0-beta.0
    Depends on vulnerable versions of lodash.pick
    node_modules/vite-plugin-checker
        @nuxt/vite-builder  >=3.0.0-rc.0
        Depends on vulnerable versions of vite
        Depends on vulnerable versions of vite-plugin-checker
        node_modules/@nuxt/vite-builder
            nuxt  >=3.9.0
            Depends on vulnerable versions of @nuxt/devtools
            Depends on vulnerable versions of @nuxt/vite-builder
            node_modules/nuxt
                @nuxt/devtools  >=1.0.7
                Depends on vulnerable versions of @nuxt/devtools-kit
                Depends on vulnerable versions of nuxt
                node_modules/@nuxt/devtools
                    @nuxt/devtools-kit  >=1.0.7
                    Depends on vulnerable versions of nuxt
                    node_modules/@nuxt/devtools-kit

vite  5.0.0 - 5.0.11
Severity: high
Vite dev server option `server.fs.deny` can be bypassed when hosted on case-insensitive filesystem - https://github.com/advisories/GHSA-c24v-8rfc-w8vw
fix available via `npm audit fix --force`
Will install nuxt@3.8.2, which is a breaking change
node_modules/@nuxt/vite-builder/node_modules/vite`

I have run npm audit fix --force which downgrades Nuxt to 3.8.3 yet it still shows the same result and cannot deploy via Netlify. 

I am using:

 `experimental: {
		componentIslands: true,	
	}, nitro: {
		preset: "netlify-edge",
	},`

and am running npx nuxi generate which was working fine earlier this week until the server.fs.deny issue became apparent. 

Btw, server components are truly brilliant - amazing work!!!

Many thanks in advance for any help/ideas.

**Comment by danielroe** [comment]
This needs to be addressed in vite.

This is a vulnerability in the vite **development server** and is not a production vulnerability.
